src/coreclr/inc/corinfo.h

```diff
@@ -2726,6 +2726,11 @@ class ICorStaticInfo
             CORINFO_CLASS_HANDLE        cls
             ) = 0;
 
+    // Returns true if a class handle represents a generic type.
+    virtual bool isGenericType(
```

Comment:
`isEnum` API that's next returns `TypeCompareState` that allows for "don't know answer". Should this and similar API follow the same style?

Reply:
You may need the "don't know answer" for `__Canon` in shared generic code.

Reply:
I assume this scenario is something akin to:

```csharp
class C<T>
{
    public bool IsGenericCode() => typeof(T).IsGenericType;
}

new C<string>().IsGenericCode();
```

Reply:
Green CI means we don't have such patterns 😄

Reply:
Fixed in https://github.com/dotnet/runtime/pull/99555/commits/85ff446faef4457d0048dd443b6a309faa58f458, added more tests in https://github.com/dotnet/runtime/pull/99555/commits/7fef10840f143c2289efed75cd4aa9adea27bfda by also including more suggestions from Egor on Discord.